pkg/utils/pipeline/results.go

```diff
@@ -0,0 +1,107 @@
+package pipeline
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"time"
+)
+
+type ResultClient struct {
+	BaseURL    string
+	HTTPClient *http.Client
+	Token      string
+}
+
+func NewClient(url, token string) *ResultClient {
+	return &ResultClient{
+		BaseURL: url,
+		HTTPClient: &http.Client{
+			Timeout: time.Minute,
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
+				},
+			},
+		},
+		Token: token,
+	}
+}
+
+func (c *ResultClient) sendRequest(path string) (body []byte, err error) {
+	requsetUrl := fmt.Sprintf("%s/%s", c.BaseURL, path)
+	fmt.Println("requestURL: ", requsetUrl)
+	req, err := http.NewRequest(http.MethodGet, requsetUrl, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+	req.Header.Set("Accept", "application/json; charset=utf-8")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.Token))
+
+	res, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	body, err = io.ReadAll(res.Body)
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to access Tekton Result Service with status code: %d and\nbody: %s", res.StatusCode, string(body))
+	}
+
+	defer res.Body.Close()
+
+	return body, err
+}
+
+func (c *ResultClient) GetRecords(namespace string, resultId string) (*Records, error) {
+	path := fmt.Sprintf("apis/results.tekton.dev/v1alpha2/parents/%s/results/%s/records", namespace, resultId)
```

Comment:
isn't there a tekton results' golang package that we could use in e2e-tests instead of using the http client?

Reply:
no, there is not.

Reply:
@psturc also I tried to find if there is tekton results pkg and couldn't find , 
maybe better we create an issue to develop this